src/meta/src/barrier/mod.rs

```diff
@@ -119,6 +119,17 @@ impl ScheduledBarriers {
             .unwrap_or_else(|| (Command::checkpoint(), Default::default()))
     }
 
+    async fn rollback_front(&self, scheduled: Scheduled) {
+        if let Command::Plain(x) = &scheduled.0 {
+            if x.is_none() {
+                return;
+            }
+        }
+        let mut buffer = self.buffer.write().await;
+        // If no command scheduled, create periodic checkpoint barrier by default.
+        buffer.push_front(scheduled);
```

Comment:
We do not ensure executing a barrier twice is correct. We have to recover the cluster currently if anything wrong happens.

Reply:
Injecting barrier happens after `update_inflight_prev_epoch`, this's only work for persisting inflight epoch, so it won't execute a barrier twice.